src/main/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -657,16 +682,21 @@ public void addInfo(final AbstractLocusInfo<SamLocusIterator.RecordAndOffset> in
                 // the raw depth may exceed the coverageCap before the high-quality depth does. So stop counting once we reach the coverage cap.
                 if (unfilteredDepth < coverageCap) {
                     unfilteredBaseQHistogramArray[recs.getRecord().getBaseQualities()[recs.getOffset()]]++;
-                    unfilteredDepth++;
                 }
+                unfilteredDepth++;
 
                 if (recs.getBaseQuality() < collectWgsMetrics.MINIMUM_BASE_QUALITY ||
                         SequenceUtil.isNoCall(recs.getReadBase()))                  { ++basesExcludedByBaseq;   continue; }
                 if (!readNames.add(recs.getRecord().getReadName()))                 { ++basesExcludedByOverlap; continue; }
-                pileupSize++;
+                highQualityDepth++;
             }
-            final int highQualityDepth = Math.min(pileupSize, coverageCap);
-            if (highQualityDepth < pileupSize) basesExcludedByCapping += pileupSize - coverageCap;
+            if (coverageCap < highQualityDepth) basesExcludedByCapping += highQualityDepth - coverageCap;
+
+            maxDepth            = Math.max(unfilteredDepth, maxDepth);
+            maxHighQualityDepth = Math.max(highQualityDepth, maxHighQualityDepth);
+
+            highQualityDepth = Math.min(highQualityDepth, coverageCap);
```

Comment:
it took me a while to see that you are indeed doing the right thing here in terms of avoiding an OOB error. 

could you help the future developer and add two new, final variables called `capped...`  for `unfilteredDepth` and `highQualityDepth`?